forge-ai/src/main/java/forge/ai/ability/PeekAndRevealAi.java

```diff
@@ -28,6 +28,11 @@ protected AiAbilityDecision checkApiLogic(Player aiPlayer, SpellAbility sa) {
             if (aiPlayer.getGame().getPhaseHandler().getPhase().isBefore(PhaseType.MAIN2)) {
                 return new AiAbilityDecision(0, AiPlayDecision.CantPlayAi);
             }
+        } else if ("AtOppEOT".equals(logic)) {
```

Comment:
I have recently pulled AtOppEOT into the parent so you no longer need to paste it into each API ;)
same is true for the other one after you switch the template into this (newer) variant

Reply:
Oh, very nice, wasn't aware of that! That's actually a really welcome change XD
I tried updating things to make use of it, seem to work fine, but please take a look if I did everything correctly.